components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -1204,6 +1204,58 @@ void RRTMGPRadiation::run_impl (const double dt) {
     });
   });
 
+  // When rad does NOT run, we fill all the rad-specific output fields (i.e., NOT T_mid)
+  // with FillValue. We also ALWAYS set extra data to signal whether rad ran or not,
+  // so that diags like ShortwaveCloudForcing can decide whether it's ok to run or not
+  // NOTE: as part of fixing https://github.com/E3SM-Project/E3SM/issues/6803, this hack
+  //       may have to be revised
+  std::vector<std::string> rad_computed_fields = {
```

Comment:
This seems like an error-prone way to do this. It would be too easy to add a computed field to rad, and then forget to (or not know to) update this list. Is it possible to query all of the computed fields for a given atm proc? That would be a safer way to do this I would think. E.g.,
```
if (update_rad) {
  ...
} else {
  std::vector<std::string> rad_computed_fields = get_computed_fields("rrtmgp");
  for (auto& name : rad_computed_fields) {
    auto f = get_field_out(name);
    f.get_header().set_extra_data("radiation_ran",update_rad);
    if (not update_rad) {
      f.deep_copy(constants::DefaultFillValue<Real>().value);
    }
  }
}